pytorch_lightning/trainer/trainer.py

```diff
@@ -2042,9 +2042,13 @@ def devices(self) -> Optional[Union[List[int], str, int]]:
 
     @property
     def data_parallel_device_ids(self) -> Optional[List[int]]:
-        return (
-            self._accelerator_connector.parallel_device_ids if self._accelerator_connector.parallel_device_ids else None
-        )
+        rank_zero_deprecation("`Trainer.data_parallel_device_ids` was deprecated in v1.6.")
```

Comment:
"and will be removed in v1.8"

Also suggest the updated alternative

Reply:
Do we still want to expose `parallel_device_ids` to users? There's no current alternative function yet. cc @four4fish

Reply:
Ideally user should just get parallel_devices instead of parallel_devices_id I think. Id can get from parallel_devices easily, it's duplicated with parallel_devices
I think after deprecate num_gpus, num_tpus and etc, we should update the device property in trainer which return the correct device for all strategies: single device return the root_device, parallel strategy return parallel_devices

Reviewers: should we provide both `devices` and `devices_id` in trainer api?